common/src/main/java/com/gentics/mesh/auth/provider/MeshJWTAuthProvider.java

```diff
@@ -44,28 +45,31 @@ public class MeshJWTAuthProvider implements AuthProvider, JWTAuth {
 
 	private static final Logger log = LoggerFactory.getLogger(MeshJWTAuthProvider.class);
 
-	private JWTAuth jwtProvider;
-
 	public static final String TOKEN_COOKIE_KEY = "mesh.token";
-
 	private static final String USERID_FIELD_NAME = "userUuid";
-
 	private static final String API_KEY_TOKEN_CODE_FIELD_NAME = "jti";
 
+	private final Vertx vertx;
 	protected final Database db;
-
-	private BCryptPasswordEncoder passwordEncoder;
-
-	private BootstrapInitializer boot;
-
+	private final BCryptPasswordEncoder passwordEncoder;
+	private final BootstrapInitializer boot;
 	private final MeshOptions meshOptions;
 
+	private JWTAuth jwtProvider;
+
 	@Inject
 	public MeshJWTAuthProvider(Vertx vertx, MeshOptions meshOptions, BCryptPasswordEncoder passwordEncoder, Database database, BootstrapInitializer boot) {
+		this.vertx = vertx;
 		this.meshOptions = meshOptions;
 		this.passwordEncoder = passwordEncoder;
 		this.db = database;
 		this.boot = boot;
+	}
+
+	public void initialize() {
+		if (this.jwtProvider != null) {
```

Comment:
can this really happen? I'd rather declare the provider as final and remove this check

Reply:
This initialization function is only needed for unit-testing.
The Provider is only getting initialized once the first request is received instead directly on startup.
In the tests this is required, as we need to replace the options once the server is already started.